samples/client/petstore/powershell_test/src/IO.Swagger/API/PetApi.ps1

```diff
@@ -0,0 +1,164 @@
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $false)]
+        [IO.Swagger.Model.Pet]
+        ${ apiKey }
+    )
+
+    Process {
+        'Calling method: PetApi-deletePet' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.deletePet(
+            ${ petId },
+            ${ apiKey }
+        )
+    }
+}
+
+function PetApi-findPetsByStatus {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ status }
+    )
+
+    Process {
+        'Calling method: PetApi-findPetsByStatus' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.findPetsByStatus(
+            ${ status }
+        )
+    }
+}
+
+function PetApi-findPetsByTags {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ tags }
+    )
+
+    Process {
+        'Calling method: PetApi-findPetsByTags' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.findPetsByTags(
+            ${ tags }
+        )
+    }
+}
+
+function PetApi-getPetById {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+    )
+
+    Process {
+        'Calling method: PetApi-getPetById' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.getPetById(
+            ${ petId }
+        )
+    }
+}
+
+function PetApi-updatePet {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ body }
+    )
+
+    Process {
+        'Calling method: PetApi-updatePet' | Write-Verbose
+        $PSBoundParameters | Out-DebugParameter | Write-Debug
+
+        $Script:PetApi.updatePet(
+            ${ body }
+        )
+    }
+}
+
+function PetApi-updatePetWithForm {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $true)]
+        [IO.Swagger.Model.Pet]
+        ${ petId }
+        [Parameter(Position = , ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true, Mandatory = $false)]
+        [IO.Swagger.Model.Pet]
+        ${ name }
```

Comment:
I used the primitive type defined in https://github.com/swagger-api/swagger-codegen/pull/5789/files#diff-f766d24f8851685d039700dd876127ceR44. Is that ok?